database/clickhouse/clickhouse.go

```diff
@@ -92,6 +92,15 @@ func (ch *ClickHouse) Open(dsn string) (database.Driver, error) {
 		migrationsTableEngine = s
 	}
 
+	// If the database is specified both in the PATH portion and the
+	// database query param of the DSN, validate that they match.
+	urlParamDB := purl.Query().Get("database")
+	urlPathDB := strings.TrimPrefix(purl.Path, "/")
+	if len(urlPathDB) != 0 && len(urlParamDB) != 0 && urlPathDB != urlParamDB {
+		return nil, fmt.Errorf("DSN string '%s' has mismatched database in path ('%s') and query parameter ('%s')",
```

Comment:
Will add a test case for this in a sec, fixing up lint issues now (can't seem to reproduce lint problems seen in CI locally)